clients/include/utility.h

```diff
@@ -6,87 +6,101 @@
-            fprintf(stderr, "rocblas_status_invalid_pointer");  \
-        }                                                       \
-        else if(error == rocblas_status_invalid_size)           \
-        {                                                       \
-            fprintf(stderr, "rocblas_status_invalid_size");     \
-        }                                                       \
-        else if(error == rocblas_status_memory_error)           \
-        {                                                       \
-            fprintf(stderr, "rocblas_status_memory_error");     \
-        }                                                       \
-        else if(error == rocblas_status_internal_error)         \
-        {                                                       \
-            fprintf(stderr, "rocblas_status_internal_error");   \
-        }                                                       \
-        else                                                    \
-        {                                                       \
-            fprintf(stderr, "rocblas_status error");            \
-        }                                                       \
-        fprintf(stderr, "\n");                                  \
-        return error;                                           \
-    }
-
-#define BLAS_1_RESULT_PRINT                       \
-    if(argus.timing)                              \
-    {                                             \
-        cout << "N, rocblas (us), ";              \
-        if(argus.norm_check)                      \
-        {                                         \
-            cout << "CPU (us), error";            \
-        }                                         \
-        cout << endl;                             \
-        cout << N << ',' << gpu_time_used << ','; \
-        if(argus.norm_check)                      \
+#define CHECK_HIP_ERROR(error)                    \
+    do                                            \
+        if(error != hipSuccess)                   \
         {                                         \
-            cout << cpu_time_used << ',';         \
-            cout << rocblas_error;                \
+            fprintf(stderr,                       \
+                    "error: '%s'(%d) at %s:%d\n", \
+                    hipGetErrorString(error),     \
+                    error,                        \
+                    __FILE__,                     \
+                    __LINE__);                    \
+            exit(EXIT_FAILURE);                   \
         }                                         \
-        cout << endl;                             \
-    }
+    while(0)
+
+#define CHECK_ROCBLAS_ERROR(error)                                  \
+    do                                                              \
+        if(error != rocblas_status_success)                         \
```

Comment:
I am not sure if you need to add extra {} for the do{} while(0)

Reply:
I added { } around it for consistency.